sqldelight-compiler/src/main/kotlin/com/squareup/sqldelight/core/compiler/DatabaseGenerator.kt

```diff
@@ -201,4 +216,18 @@ internal class DatabaseGenerator(
         .primaryConstructor(constructor.build())
         .build()
   }
+
+  private fun SqlStmt.isSchema() = when {
+    createIndexStmt != null -> true
+    createTableStmt != null -> true
+    createTriggerStmt != null -> true
+    createViewStmt != null -> true
+    createVirtualTableStmt != null -> true
+    alterTableStmt != null -> true
+    dropIndexStmt != null -> true
+    dropTableStmt != null -> true
+    dropTriggerStmt != null -> true
+    dropViewStmt != null -> true
+    else -> false
+  }
```

Comment:
Interesting take. This is basically a chained `&&`. I assume the bytecode is the same too. Kinda weird, but I don't care.

Reply:
i hate long if statements, this feels more readable and maintainable